modules/core/src/main/java/org/apache/ignite/events/QueryExecutionEvent.java

```diff
@@ -0,0 +1,155 @@
+ *          asynchronously querying events occurred on the nodes specified, including remote nodes.
+ *      </li>
+ *      <li>
+ *          {@link org.apache.ignite.IgniteEvents#localQuery(org.apache.ignite.lang.IgnitePredicate, int...)} -
+ *          querying only local events stored on this local node.
+ *      </li>
+ *      <li>
+ *          {@link org.apache.ignite.IgniteEvents#localListen(org.apache.ignite.lang.IgnitePredicate, int...)} -
+ *          listening to local grid events (events from remote nodes not included).
+ *      </li>
+ * </ul>
+ * User can also wait for events using method {@link org.apache.ignite.IgniteEvents#waitForLocal(org.apache.ignite.lang.IgnitePredicate, int...)}.
+ * <h1 class="header">Events and Performance</h1>
+ * Note that by default all events in Ignite are enabled and therefore generated and stored
+ * by whatever event storage SPI is configured. Ignite can and often does generate thousands events per seconds
+ * under the load and therefore it creates a significant additional load on the system. If these events are
+ * not needed by the application this load is unnecessary and leads to significant performance degradation.
+ * <p>
+ * It is <b>highly recommended</b> to enable only those events that your application logic requires
+ * by using {@link org.apache.ignite.configuration.IgniteConfiguration#getIncludeEventTypes()} method in Ignite configuration. Note that certain
+ * events are required for Ignite's internal operations and such events will still be generated but not stored by
+ * event storage SPI if they are disabled in Ignite configuration.
+ *
+ * @see EventType#EVT_QUERY_EXECUTION
+ */
+public class QueryExecutionEvent<K, V> extends EventAdapter {
+    /** */
+    private static final long serialVersionUID = 3738753361235304496L;
+
+    /** Query type. */
+    private final String qryType;
+
+    /** Clause. */
+    private final String clause;
+
+    /** Query arguments. */
+    @GridToStringInclude
+    private final Object[] args;
+
+    /** Security subject ID. */
+    private final UUID subjId;
+
+    /**
+     * @param node Node where event was fired.
+     * @param msg Event message.
+     * @param type Event type.
+     * @param qryType Query type.
+     * @param clause Clause.
+     * @param args Query arguments.
+     * @param subjId Security subject ID.
+     */
+    public QueryExecutionEvent(
+        ClusterNode node,
+        String msg,
+        int type,
+        String qryType,
+        @Nullable String clause,
+        @Nullable Object[] args,
+        @Nullable UUID subjId
+    ) {
+        super(node, msg, type);
+
+        assert qryType != null;
+
+        this.qryType = qryType;
+        this.clause = clause;
+        this.args = args;
+        this.subjId = subjId;
+    }
+
+    /**
+     * Gets query type.
+     *
+     * @return Query type. Can be {@code "SQL"}, {@code "SQL_FIELDS"}, {@code "FULL_TEXT"}, {@code "SCAN"},
```

Comment:
I see trigger only for SQL_FIELDS event. What about others?

Reply:
Added.

Reply:
I think we should trigger only SQL_FIELDS and SQL types. Other types are not about SQL.

Reply:
But we fire event for any query, isn't it?

Reply:
@joooger @SomeFire hi, is there some understanding?

Reply:
@joooger, I removed other types as you suggested.

Reply:
I still see all types in Java Doc :)

Reply:
also, what about SQL type?

Reply:
Actually I don't understand why others queries are skipped? Is SCAN query differs from SQL from this point of view?

Reply:
`querySql()` have `querySqlFields()` inside, so, no additions needed.

I'll fix docs in a few minutes.